### PR TITLE
Add Google auth to allow users to store data across accounts

### PR DESCRIPTION
**Pull request by sophia-hubscher**
This PR adds Google authentication to ClosetIQ, enabling users to log in and store data across accounts. Key updates include:  

1. **Google Auth Integration**:  
   - Made the login login button in the navigation bar use Google authentication.  

2. **Front-End Utility Functions**:  
   - Created `getId` and `getUsername` functions to retrieve the authenticated user's ID and username. These functions can be reused across the front-end for consistent user data management.  

3. **Documentation Update**:  
   - Updated the README to reflect the new authentication feature and provide guidance for setup.  

### Next Steps  
In subsequent PRs, the team should use the newly added `getId` and `getUsername` functions to ensure that objects are correctly stored and fetched for the authenticated user.  

### Video Demo
https://github.com/user-attachments/assets/6faa7dd2-4fe7-4e5a-a071-57157c137b44

### CODE REVIEW

**Comment by sophia-hubscher** [comment]
**Update:** All pages are now hidden until the user is logged in

File: back-end/package.json
Comment (nhan0504) [inline back-end/package.json:1]: In the file `passport.js` it's using `dotenv` package but I don't see that package in package.json. It was giving me error so I installed `dotenv`. So adding this to  package.json  `"dotenv": "^16.4.7",`

**Review by nhan0504 — CHANGES_REQUESTED** [review]
Just some package issue but other LGTM!

**Review by nhan0504 — APPROVED** [review]
LGTM!

**Pull request merged**